src/openzeppelin/utils/constants/library.cairo

```diff
@@ -1,36 +1,36 @@
-# SPDX-License-Identifier: MIT
-# OpenZeppelin Contracts for Cairo v0.3.2 (utils/constants/library.cairo)
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts for Cairo v0.3.2 (utils/constants/library.cairo)
 
 %lang starknet
 
-#
-# Numbers
-#
+//
+// Numbers
+//
 
-const UINT8_MAX = 256
+const UINT8_MAX = 256;
 
-#
-# Interface Ids
-#
+//
+// Interface Ids
+//
 
-# ERC165
-const IERC165_ID = 0x01ffc9a7
-const INVALID_ID = 0xffffffff
+// ERC165
+const IERC165_ID = 0x01ffc9a7;
+const INVALID_ID = 0xffffffff;
 
-# Account
-const IACCOUNT_ID = 0xf10dbd44
+// Account
+const IACCOUNT_ID = 0xf10dbd44;
```

Comment:
We should update that value according to the new `IAccount` interface.

Reply:
Yes, good call.

Reply:
@juniset @martriay we're including `isValidSignature` in the interface, correct? If so, this is how it looks. Let me know if there are any objections.

```
interface IAccount {
    function isValidSignature(bytes32, bytes32, bytes32) external;    // hash, r, s
    function __execute__(address, bytes4, bytes calldata) external;   // to, selector, calldata
    function __validate__(address, bytes4, bytes calldata) external;  // to, selector, calldata
    function __validate_declare__(bytes32) external;                  // class hash
}
```

Reply:
By the way, shouldn't we include `supportsInterface` to the `IAccount` interface?

Reply:
@juniset right! Thank you, that's much better^.

Regarding the id calculation, does it make sense to use `uint256[] data` instead of `uint256 data` in `Call`?

Reply:
You're right.But I think it should be `bytes` then.

Reply:
Yes, `bytes` is better. So, here's the updated interface:

```
interface IAccount {

    struct Call {
        address to;
        bytes4 selector;
        bytes data;
    }

    function isValidSignature(bytes32, bytes32[] calldata) external;
    function __execute__(Call[] calldata) external;
    function __validate__(Call[] calldata) external;
    function __validate_declare__(bytes32) external;
}

contract Selector {

    // returns 0xa7941cd2
    function calculateSelector() public pure returns (bytes4) {
        IAccount i;
        return i.isValidSignature.selector ^
        i.__execute__.selector ^
        i.__validate__.selector ^
        i.__validate_declare__.selector;
    }
}
```

which gives us `IACCOUNT = 0xa7941cd2`.

Reply:
Perfect. And I just spoke to @martriay and we're all in agreement with adding `supportsInterface` to `IAccount`; therefore:

```
interface IAccount {

    struct Call {
        address to;
        bytes4 selector;
        bytes data;
    }

    function supportsInterface(bytes4) external;
    function isValidSignature(bytes32, bytes32[] calldata) external;
    function __execute__(Call[] calldata) external;
    function __validate__(Call[] calldata) external;
    function __validate_declare__(bytes32) external;
}

contract Selector {

    // returns 0xa66bd575
    function calculateSelector() public pure returns (bytes4) {
        IAccount i;
        return i.supportsInterface.selector^
        i.isValidSignature.selector ^
        i.__execute__.selector ^
        i.__validate__.selector ^
        i.__validate_declare__.selector;
    }
}
```

which gives us `IACCOUNT = 0xa66bd575`. What does everyone think?

Reply:
Thanks @andrew-fleming , that looks good to me!

Reply:
Awesome! I still have two doubts:

- should `bytes4` be `bytes32` considering that selectors are way longer (`1 felt`) in Cairo?
- our current interface doesn't actually take `Call[]` because we can't handle arrays of complex structs yet. I guess we want to be future proof but do we agree this isn't the current interface?

other than that, LGTM

Reply:
> should bytes4 be bytes32 considering that selectors are way longer (1 felt) in Cairo?

I agree that `bytes32` is a more accurate type for SN selectors; however, `bytes4` seems more compliant with EIP165. I vote for keeping `bytes4`

> our current interface doesn't actually take Call[] because we can't handle arrays of complex structs yet. I guess we want to be future proof but do we agree this isn't the current interface?

And yes, I agree the proposed interface is the interface-to-be, not the interface right now